CppCoreGuidelines.md

```diff
@@ -2766,24 +2770,45 @@ Passing a shared smart pointer (e.g., `std::shared_ptr`) implies a run-time cost
         // ...
     };
 
-See further in [R.30](#Rr-smartptrparam).
+    // caller
+    shared_ptr<widget> my_widget = /* ... */;
+    f(my_widget);
 
-##### Note
+    widget stack_widget;
+    f(stack_widget); // error
 
-We can catch dangling pointers statically, so we don't need to rely on resource management to avoid violations from dangling pointers.
+##### Example, good
 
-**See also**:
+    // callee
+    void f(widget& w)
+    {
+        // ...
+        use(w);
+        // ...
+    };
 
-* [Prefer `T*` over `T&` when "no argument" is a valid option](#Rf-ptr-ref)
-* [Smart pointer rule summary](#Rr-summary-smartptrs)
+    // caller
+    shared_ptr<widget> my_widget = /* ... */;
+    f(*my_widget);
+
+    widget stack_widget;
+    f(stack_widget); // ok -- now this works
+
+##### Note
+
+We can catch dangling pointers statically, so we don't need to rely on resource management to avoid violations from dangling pointers.
 
 ##### Enforcement
 
-Flag a parameter of a smart pointer type (a type that overloads `operator->` or `operator*`) for which the ownership semantics are not used;
```

Comment:
I'm using the R.30 wording over F.7 wording for the enforcement - I think R.30's is spelled in a more actionable way.